accepted/single-file/design.md

```diff
@@ -66,225 +57,204 @@ Publishing to a single file can be triggered by adding the following property to
+
+* ReadyToRun [Composite](https://github.com/dotnet/runtime/blob/master/docs/design/features/readytorun-composite-format-design.md) Assemblies are expected to be loaded similar to ReadyToRun assemblies. 
+
+#### Windows Limitations
+The Windows mapping routines have the following limitations:
+* [CreateFileMapping](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createfilemappinga) has no option to create a mapping for a part of the file (no offset argument).
+    This means that:
+    * We cannot use the (SEC_IMAGE) attribute to perform automatic section-wise loading (circumventing alignment requirements) of bundled assemblies directly.
+    * Instead we need to map each section independently.
+* [MapViewOfFile]( https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-mapviewoffileex) can only map parts of a file aligned at [memory allocation granularity]( https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/ns-sysinfoapi-system_info), which is 64KB.
+    This means that each section within the assemblies should be aligned at 64KB – which is not guaranteed by the crossgen compiler.
+
+We therefore map the assembly and perform an in-memory copy of the sections to appropriate offsets. In the long term, the solution to the mapping problem would involve considerations such as:
+* Compile all assemblies in a version bubble into one PE assembly, with a few aligned sections.
+* Embed the big composite assembly into the host with proper alignment, so that the single-exe bundle can be loaded without copies at run time.
+
+### API Semantics
 
 #### `Assembly.Location`
 
-There are a few options to consider for the `Assembly.Location`  property of a bundled assembly:
+There are a few options to consider for the `Assembly.Location` property of a bundled assembly:
 
 * A fixed literal (ex: `null`) indicating that no actual location is available.
 * The simple name of the assembly (with no path).
-* A special UNC notation such as `<bundle-path>/:/asm.dll` to denote files that come from the bundle. 
 * The path of the assembly as if it were not to be packaged into the single-file.
-* In the case of files spilled to disk (ex: ready-to-run assemblies) -- the actual location of the extracted file.
+* A special UNC notation such as `<bundle-path>/:/asm.dll` to denote files that come from the bundle. 
 * A configurable selection of the above, etc.
 
-We propose keeping the default behavior of `Assembly.Location`. That is, 
-
-* If the assembly is loaded directly from the bundle, return empty-string
-* If the assembly is extracted to disk, return the location of the extracted file.
+Proposed solution is for `Assembly.Location` to return `null` for bundled assemblies, which is the default behavior for assemblies loaded from memory.
```

Comment:
I believe this should actually be an empty string, not null, as per the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.location?view=netframework-4.8) (I came back to this while helping .NET Native customer with yet another issue caused by this).

Reply:
Thanks @MichalStrehovsky, I'll fix this.